src/notation/CMakeLists.txt

```
@@ -242,4 +242,8 @@ if (NOT MSVC AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VER
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-copy")
 endif (NOT MSVC AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0)
 
+if (MSVC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj")
```

```suggestion
    add_compile_options("/bigobj")
```
or
```suggestion
    target_compile_options(notation PRIVATE "/bigobj")
```
but then it needs to be after `setup_module()`.

Or you could decide to make it global, and put the `add_compile_options` version in `SetupBuildEnvironment`. Perhaps that's best, but I'm not very sure. I don't know how big the downsides of this `/bigobj` option are. (If it's only that we can't use pre-2005 linkers with it, then we would surely be able to live with it, but it's unclear whether that's the whole story.)

By the way, I guess that the root cause is that the files become too large because of unity build.

---

Well, I copied the code from https://github.com/musescore/MuseScore/pull/21018/files#diff-13e09e8d26017906f5940595da8c1ecb23b99fa9e655d67625b72f882c0cbd48R45-R47

But yes, this does seem related to Unity builds

---

Okay, I'll merge it, and then we can investigate improved versions later.